sound/soc/sof/pm.c

```diff
@@ -229,9 +229,19 @@ static void sof_suspend_streams(struct snd_sof_dev *sdev)
 static int sof_resume(struct device *dev, bool runtime_resume)
 {
 	struct sof_platform_priv *priv = dev_get_drvdata(dev);
-	struct snd_sof_dev *sdev = dev_get_drvdata(&priv->pdev_pcm->dev);
+	struct snd_sof_dev *sdev;
 	int ret = 0;
 
+	if (!priv) {
+		dev_dbg(dev, "%s: priv is NULL", __func__);
+		return 0;
```

Comment:
Can this really happen? the sof_pci_probe function starts with 

````c
static int sof_pci_probe(struct pci_dev *pci,
			 const struct pci_device_id *pci_id)
{
[...]
	dev_dbg(&pci->dev, "PCI DSP detected");

	/* get ops for platform */
	ops = desc->ops;
	if (!ops) {
		dev_err(dev, "error: no matching PCI descriptor ops\n");
		return -ENODEV;
	}

	priv = devm_kzalloc(dev, sizeof(*priv), GFP_KERNEL);
	if (!priv)
		return -ENOMEM;

	pci_set_drvdata(pci, priv);
````
so if "priv" was ever NULL we would not even be in the PM stuff.

Reply:
@plbossart 
Yes, you are right. This should never happen logically. I add this code just for sanity check in case our code is wrong...
If you don't like it, I will remove this check to make the code simple.